public/js/livesearch.js

```diff
@@ -0,0 +1,104 @@
+	//init vars
+	var input;
+	var results;
+	var indiResults;	
+	var urlResult;
+	var formMethod;
+	var typeData;
+	var resultFilter;
+	var identifier ;
+
+	//set values to vars, this is done wherever the livesearch is called
+	function setInputListener(inputField , resultField, resultHolder, uUrl, method, typeD){
+		
+		input = inputField;
+		results = resultField;
+		indiResults = resultHolder;
+		urlResult =  uUrl + '/';
+		typeData = typeD;
+		return this;
+	}	
+	
+	//set the method type for ajax submission
+	function setMethod(method){
+		var regex= /(GET|PUT)/g;
+		var methods = method.match(regex);
+		if(methods != null){
+			return this;
+		}
+		console.log("The method was not set correctly");
+		return false;
+	}
+
+	//setting the filter / find for returning only desired result html
+	function setResultFilter(filter){
+		resultFilter = filter;
+		return this;
+	}
+	
+	//debugging
+	//console.log("live search is loaded");
+
+	//actual jquery live search fn, takes in the form where live search is being performed
+	(function( $form ){
+		$.fn.liveSearch = function() {	
+			// debugging
+			// console.log($(this));
+			
+			//giving the form it's own var for simplicity sake, later functions redefine the 'this' method 
+			var uForm = $(this);
+			
+			//taking cleaning user supplied input - no extra paces or , will be queried
+			function filterResults(raw){
+				var filtered = raw.replace(/\s+/g,' ').trim().replace(',','');
+				return filtered;
+			}
+			//set the form input element text to cleaned text, nice and tidy
+			$(this).find(":text").val(filterResults($(this).find(":text").first().val()));
+			var userInput=  $(this).find(":text").val();
+			
+			//ajax - asynchronously requesting resutls from our result page
+			$.ajax({
+			type: formMethod,
+			url: urlResult+userInput,
+			dataType: typeData,
+			data: $(this).serialize(),
+			success:function(data){
+				//clear the search result display each time we look up new results - otherwise duplicates show
+				results.empty();
+				var userResults = $(data).find(resultFilter);
+				//debugging
+				//console.log(arrayReturend);
+				
+				//jquery for each short hand, populating result list with well...results
+				$(userResults).each(function(){
+					console.log($(this));
+					results.append(indiResults+ $(this).text()+ "</li>");
```

Comment:
Appending the `"</li>"`  to the end of the string assumes that everyone passed in an `<li>` element for `resultHolder` which may not be true. If it is always true then you might want to just remove the option to set the resultHolder and always make it the `<li class='list-group-item suggestion'>` element you passed in.